File: src/main/java/dev/slne/spawn/trader/manager/object/impl/GlobeTrade.java
package dev.slne.spawn.trader.manager.object.impl;

import dev.slne.spawn.trader.manager.object.Trade;
import dev.slne.spawn.trader.util.ItemBuilder;
import it.unimi.dsi.fastutil.objects.ObjectList;
import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;

public class GlobeTrade implements Trade {
    @Override
    public ObjectList<ItemStack> requirements() {
        return ObjectList.of(new ItemBuilder(Material.EMERALD, 5).build(), new ItemBuilder(Material.PAPER, 20).build());
    }

    @Override
    public ObjectList<ItemStack> rewards() {
        return ObjectList.of(new ItemBuilder(Material.GLOBE_BANNER_PATTERN).build());
    }

    @Override
    public Integer id() {
        return 2;
    }

    @Override
    public String name() {
        return "globe-banner-pattern";
    }

    @Override
    public String rewardMessage() {
        return "Danke f\u00FCr den Einkauf von einer Globe Banner Vorlage!";
    }
}

File: src/main/java/dev/slne/spawn/trader/manager/trade/impl/GlobeTrade.kt
package dev.slne.spawn.trader.manager.trade.impl

import dev.slne.spawn.trader.manager.trade.Trade
import dev.slne.surf.surfapi.bukkit.api.builder.buildItem
import dev.slne.surf.surfapi.core.api.util.mutableObjectListOf
import it.unimi.dsi.fastutil.objects.ObjectList
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class GlobeTrade : Trade {
    override fun requirements() = mutableObjectListOf(
        buildItem(Material.EMERALD, 20) {},
        buildItem(Material.PAPER, 20) {}
    )

    override fun rewards(): ObjectList<ItemStack> {
        return mutableObjectListOf(buildItem(Material.GLOBE_BANNER_PATTERN) {})
    }

    override fun id(): Int {
        return 2
    }

    override fun name(): String {
        return "globe-banner-pattern"
    }

    override fun rewardMessage(): String {
        return "Danke für den Einkauf von einer Globe Banner Vorlage!"
    }

    override fun cooldown(): Long {
        return 75600000L
    }
}
